crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -576,24 +579,35 @@ async fn validate_and_publish_software_list(
     Ok(vec![])
 }
 
-async fn execute_operation(payload: &str, command: &str) -> Result<(), CumulocityMapperError> {
+async fn execute_operation(
+    payload: &str,
+    command: &str,
+    operation_name: &str,
+) -> Result<(), CumulocityMapperError> {
     let command = command.to_owned();
     let payload = payload.to_string();
 
-    let _handle = tokio::spawn(async move {
```

Comment:
Removing this `tokio::spawn` makes the operation blocking and will prevent the mapper to do anything till the operation is ended.

It's correct to do as done here . i.e to spawn a child process to be able to check an error on launch. But, if the process is successfully spawned, one needs to wait for the child to terminate in a `tokio::spawn` thread.

=> The `tokio::spawn` removed here must be re-introduced elsewhere.

Reply:
Oh! Thank you, I did wonder why that was there. It seemed redundant, but now I see how important it is!